### PR TITLE
Improve code coverage in kubectl annotate

### PR DESCRIPTION
**Pull request by weikhor**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
To have more code coverage in annotate test

1) Master branch
- ok      k8s.io/kubectl/pkg/cmd/annotate 0.067s  coverage: 73.8% of statements

2) Local branch, weikhor:annotate_code_coverage
- ok      k8s.io/kubectl/pkg/cmd/annotate 0.067s  coverage: 80.2% of statements

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?
No

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by mpuckett159** [comment]
@weikhor Can you please get the CLA signed? For information please see this guide.

https://github.com/kubernetes/community/blob/master/CLA.md

**Comment by weikhor** [comment]
> @weikhor Can you please get the CLA signed? For information please see this guide.
> 
> https://github.com/kubernetes/community/blob/master/CLA.md

Yes. I not sure why this happens. I do not face this problem in my first PR, I already submit support CLA ticket.

**Comment by mpuckett159** [comment]
Awesome, thank you.

/ok-to-test

**Comment by mpuckett159** [comment]
Nice, thanks for this 😄 

/lgtm

File: staging/src/k8s.io/kubectl/pkg/cmd/annotate/annotate_test.go [hunk 1]
```diff
@@ -474,18 +474,57 @@ func TestAnnotateObject(t *testing.T) {
...
+func TestAnnotateObjectMultipleResources(t *testing.T) {
+	pods, _, _ := cmdtesting.TestData()
+
+	tf := cmdtesting.NewTestFactory().WithNamespace("test")
+	defer tf.Cleanup()
+
+	codec := scheme.Codecs.LegacyCodec(scheme.Scheme.PrioritizedVersionsAllGroups()...)
+
+	tf.UnstructuredClient = &fake.RESTClient{
+		GroupVersion:         schema.GroupVersion{Group: "testgroup", Version: "v1"},
```
Comment (brianpursley) [inline staging/src/k8s.io/kubectl/pkg/cmd/annotate/annotate_test.go:518]: Do we need this line?  
```
GroupVersion: schema.GroupVersion{Group: "testgroup", Version: "v1"},
```

I don't think this unit test does anything with testgroup/v1 resources.

Same goes for line 473.

**Review by brianpursley — CHANGES_REQUESTED** [review]
Hi @weikhor.  Thanks for adding this unit test!

I notice that `TestAnnotateObject` is similar in structure to `TestAnnotateObjectMultipleResources` and I wonder if they can be combined into a single test function with multiple cases.

See, for example, how [`TestAnnotateErrors`](https://github.com/kubernetes/kubernetes/blob/fbd45fe7c12c87fab144c6dcc885e97b0e962d55/staging/src/k8s.io/kubectl/pkg/cmd/annotate/annotate_test.go#L400-L434) does it.

---
Also, I think this would be a good opportunity to improve the existing `TestAnnotateObject` unit test, since you are touching that code.

I notice that the test currently only checks that the `RunAnnotate()` returns nil and doesn't really assert anything to make sure the actual behavior is what we expect.

I can replace the entire code of `RunAnnotate` with this, and the test still passes:
```
func (o AnnotateOptions) RunAnnotate() error {
	return nil
}
```

So my question is, how can we assert the behavior of RunAnnotate?

So let's take these args for example:
```
args := []string{"pods/foo", "a=b", "c-"}
```

I believe this adds an annotation called `a` with a value of `b` and remove annotation `c`, right?

I wonder if there is a way in the fake HTTP Client that we can assert that the request body matches what we would expect for those changes to be made?

Maybe we could have expected `expectedPatchBody` and `actualPatchBody` variables, and at the end, after `RunAnnotate()` completes, then check to make sure `actualPatchBody == expectedPatchBody`?

File: staging/src/k8s.io/kubectl/pkg/cmd/annotate/annotate_test.go (hunk 1 above)
Comment (weikhor) [reply staging/src/k8s.io/kubectl/pkg/cmd/annotate/annotate_test.go]: Noted. Thank. I will update accordingly

**Comment by mpuckett159** [comment]
> I believe this adds an annotation called `a` with a value of `b` and remove annotation `c`, right?
> 
> I wonder if there is a way in the fake HTTP Client that we can assert that the request body matches what we would expect for those changes to be made?
> 
> Maybe we could have expected `expectedPatchBody` and `actualPatchBody` variables, and at the end, after `RunAnnotate()` completes, then check to make sure `actualPatchBody == expectedPatchBody`?

@brianpursley I think this would probably be best served by unit tests for the `updateAnnotations` function as it is what actually creates the request body, and this is already covered by the `TestUpdateAnnotations` test from my reading of the tests.

**Pull request closed without merging**